File: server/php/src/Application/Domain/Payments/PaymentDetails/SepaDirectDebitPaymentMethodSpecificOutput.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails;

final class SepaDirectDebitPaymentMethodSpecificOutput
{
    public ?FraudResults $fraudResults = null;
    public ?PaymentProduct771SpecificOutput $paymentProduct771SpecificOutput = null;
    public ?int $paymentProductId = null;

    public function __construct(
        ?int $paymentProductId = null,
        ?FraudResults $fraudResults = null,
        ?PaymentProduct771SpecificOutput $paymentProduct771SpecificOutput = null
    ) {
        $this->paymentProductId = $paymentProductId;
        $this->fraudResults = $fraudResults;
        $this->paymentProduct771SpecificOutput = $paymentProduct771SpecificOutput;
    }

    public function toArray(): array
    {
        return [
            'paymentProductId' => $this->paymentProductId,
            'fraudResults' => ($this->fraudResults ?? new FraudResults())->toArray(),
            'paymentProduct771SpecificOutput' => ($this->paymentProduct771SpecificOutput ?? new PaymentProduct771SpecificOutput())->toArray(),
        ];
    }
}

File: server/python/app/application/domain/payments/payment_details/sepa_direct_debit_payment_method_specific_output.py
from dataclasses import dataclass
from typing import Optional

from app.application.domain.payments.payment_details.fraud_result import FraudResults
from app.application.domain.payments.payment_details.payment_product_771_specific_output import PaymentProduct771SpecificOutput

@dataclass
class SepaDirectDebitPaymentMethodSpecificOutput:
    fraud_results: Optional[FraudResults] = None
    payment_product_771_specific_output: Optional[PaymentProduct771SpecificOutput] = None
    payment_product_id: Optional[int] = None